miden-crypto/src/dsa/ecdsa_secp256k1/key_agreement.rs

```diff
@@ -0,0 +1,121 @@
+//! ECDH (Elliptic Curve Diffie-Hellman) key agreement implementation over secp256k1 curve.
+//!
+//! Note that the intended use is in the context of a one-way, sender initiated key agreement
+//! scenario. Namely, when the sender knows the (static) public key of the receiver and it
+//! uses that, together with an ephemeral secret key that it generates, to derive a shared
+//! secret.
+//! This shared secret will then be used to encrypt some message (using for example a key
+//! derivation function).
+//! The public key associated with the ephemeral secret key will be sent alongside the encrypted
+//! message.
+
+use hkdf::{Hkdf, hmac::SimpleHmac};
+use k256::{
+    AffinePoint,
+    elliptic_curve::rand_core::{CryptoRng, RngCore},
+    sha2::Sha256,
+};
+
+use super::PublicKey;
+
+/// A shared secret computed using the ECDH (Elliptic Curve Diffie-Hellman) key agreement.
+pub struct SharedSecret {
+    inner: k256::ecdh::SharedSecret,
+}
+
+impl SharedSecret {
```

Comment:
We probably will need to reduce this to a `Word` (e.g., for RPO-based encryption). Should we add a `to_word()` method for this too?

Reply:
What we can do, if we can, is to use the `extract method` to do this. Do you see anything that can block this?